File: 11.container-with-most-water.cpp
/*
 * @lc app=leetcode id=11 lang=cpp
 *
 * [11] Container With Most Water
 */

// @lc code=start
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i=0;
        int j=height.size()-1;
        int s=0;
        while (i<j){
            s=max(s,min(height[i],height[j])*(j-i));
            if (height[i]>height[j])
            j--;
            else 
            i++;
        }
        return s;
    }
};
/*双指针法// @lc code=end

File: 11.container-with-most-water.py
#
# @lc app=leetcode id=11 lang=python3
#
# [11] Container With Most Water
#

# @lc code=start
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i=0
        j=len(height)-1
        s=0
        while i<j:
            s=max(s,min(height[i],height[j])*(j-i))
            if height[i]<height[j]:
                i+=1
            else:
                j-=1
        return s
# @lc code=end

'''原理同样是双指针